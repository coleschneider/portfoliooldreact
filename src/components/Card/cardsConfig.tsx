import BlendLogo from '../../assets/images/BlendLogo.png'
import BlendLogoSmall from '../../assets/images/BlendLogoSmall.jpg'
import DanMillerATI from '../../assets/images/DanMillerATI.png'
import DanMillerATISmall from '../../assets/images/DanMillerATISmaller.jpg'
import ORCASPreview from '../../assets/images/ORCASPreview.png'
import ORCASPreviewSmall from '../../assets/images/ORCASPreviewSmall.jpg'

export const cardsConfig: Card[] = [
  {
    id: 'blend',
    description: 'Blend',
    cardImage: BlendLogo,
    placeholder: BlendLogoSmall,
    position: 'Software Engineer (Intern)',
    link: 'https://blend.com',
    details: [
      {
        title: `Blend (Software Engineer Internship)`,
        body: `I took a semester off at Pepperdine to intern at Blend from September through December. I worked with the engineering team on multiple projects frontent and backend. All projects were 100% covered by unit, integration, and E2E tests. The projects I worked on improving and implementing new login workflows for various clients, banks, and financial instutions.`,
      },
      {
        title: `Development`,
        body: `At Blend, I communicated with project managers to achieve goals set during sprint planning and followed up with these goals during stand up. I also worked with design, product, and legal teams to meet deadlines in ways that distributed engineer workload and preserved scalability of the app. I refactored banking login apps for improved routing and mobile responsive layout. I also built a custom frontend tracking system that reported events to improve analytics.`,
      },
      {
        title: `Build Speed`,
        body: `At Blend, I worked on the "borrower" team for lending. Because we managed quite a bit of consumer-facing side of the app, we wanted to ensure quick loading times and an overall positive experience. The first thing I did to improve the build speed was upgrade Blend's webpack version from 2 to 4 so that we could take advantage of code-splitting and lazy-loading. The second thing I did was selectively import polyfills depending on the type of browser the user was on. The last thing I did was split the vendor assets into seperate chunks so that user's didn't download all of the app's assets on the initial page load.`,
      },
      {
        title: `Testing`,
        body: `Achieved 100% coverage with jest and enzyme in all frontend unit and integration tests. New workflows I added followed existing patterns where inputs sent from the backend were transformed, validated, and displayed on the frontend app through schemas. This also included 100% testing with mocha.`,
      },
    ],
  },
  {
    id: 'orca-seniors',
    description: 'ORCA Seniors',
    position: 'Developer for a non-profit',
    cardImage: ORCASPreview,
    placeholder: ORCASPreviewSmall,
    link: 'https://orcaseniors.org',
    details: [
      {
        title: `ORCA Seniors`,
        body: `During my first Senior semester at Pepperdine University, I worked with a team of other students to improve the current state of a non profit organization. We we're assigned the ORCAS who's mission is to build a better senior center for the community of Oxnard, CA. My team and I presented them with multiple new strategies to help achieve their goal, one of the being a new website that I woked on for the duration of the project.`,
      },
      {
        title: `Development`,
        body: `I developed the site using React, emphasizing the importance of the ORCAs goals and collaborating with them to ensure they ended up with a product that fit their mission, vision, and values. This was the first progressive site I had worked on by myself and had a lot of fun implementing offline capabilities of service workers. I focused on alerting the user with various prompts so navigating the app was simple and also directed. I also implemented the email eventing system that subscribes users through MailChimp by creating various AWS Lambda functions.`,
      },
      {
        title: `Testing`,
        body: `For testing the frontend code, I used jest and enzyme. I wrote unit-tests for each component and integration tests for higher level testing. I tested the AWS Lambda functions that sent newsletter information with mocha, and tied both suites of testing together to make sure one did not deploy without the other working properly.`,
      },
      {
        title: `Deployment`,
        body: `Because my team was working closely with the ORCAs on a daily basis, I wanted to make sure that I could make quick and safe changes to the app through continuous deployment. I set up two S3 buckets, one for logs and another to store the static assets from the webpack build. Then, I set up CircleCI build process that ran tests, built the app, and deployed it in parallel rather than running these steps sequentially. One issue I ran into with this was that Cloudfront was not properly invalidating the cache of assets in the S3 bucket and the service worker installed did not update. After looking into it, the solution was to simply set the max-age of new assets on the S3 bucket.`,
      },
    ],
  },
  {
    id: 'dan-miller-ati',
    description: 'Dan Miller ATI',
    position: 'Developer / Project',
    cardImage: DanMillerATI,
    placeholder: DanMillerATISmall,
    link: 'https://www.danmillerati.com',
    details: [
      {
        title: `Dan Miller ATI`,
        body: `This was a really fun site I got to work on for a friend that showcased his skill with auto, boat, and aircraft upholstery.`,
      },
      {
        title: `Development`,
        body: `The project was really fun because he had so many images that displayed his work and we didn't want to exclude any of them. The biggest issue I ran into was displaying all the images on his site in a way that didn't compromise the user experience. I ended up lazy loading all the images by measuring the dimensions of the user's screen relative to how far they scrolled down the page. I wrapped each image in a promise that resolved with the image's url and cached it by taking advantage of React's "useReducer" function. Then, in order to avoid passing props to deeply, I passed the state of the reducer to a context provider so that I could access the image's loading status from anywhere within the app.`,
      },
      {
        title: `Testing & Deployment`,
        body: `I used jest and enzyme to test the gallery component and the rest of the app. I used CircleCI to run test, build, and deployment tasks to ensure code quality through linting and formatting pre-deployment.`,
      },
    ],
  },
]

export const cardsById = cardsConfig.reduce(
  (acc, curr) => {
    acc[curr.id.toString()] = curr
    return acc
  },
  {} as {
    [k: string]: Card
  },
)

// export default cardsConfig
