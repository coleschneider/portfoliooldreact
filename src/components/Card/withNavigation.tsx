import React from 'react'

// function withNavigation(Component) {
//   return class Navigation extends React.Component {
//     element = null

//     componentDidMount() {
//       window.addEventListener('resize', this.handleResize)
//     }

//     componentWillUnmount() {
//       window.addEventListener('resize', this.handleResize)
//     }

//     handleResize = () => {
//       if (this.element) {
//         const { top, right, bottom, left, width, height } = this.element.getBoundingClientRect()
//         this.props.onUpdateCards({ top, right, bottom, left, width, height }, this.props.id)
//       }
//     }

//     render() {
//       const { onUpdateCards, history, id, ...rest } = this.props
//       const goToElement = () => {
//         const { top, right, bottom, left, width, height } = this.element.getBoundingClientRect()
//         this.props.onUpdateCards({ top, right, bottom, left, width, height }, this.props.id)
//         history.push({
//           pathname: `/work/${id}`,
//           state: {
//             id,
//             to: 'modal',
//             meta: {
//               from: { top, right, bottom, left, width, height },
//             },
//           },
//         })
//       }
//       return (
//         <div ref={el => (this.element = el)}>
//           <Component onClick={goToElement} id={id} {...rest} />
//         </div>
//       )
//     }
//   }
// }
function withNavigation(Component) {
  return function Navigation(props) {
    const element = React.useRef(null)
    React.useEffect(() => {
      const handleResize = () => {
        if (element.current) {
          const { top, right, bottom, left, width, height } = element.current.getBoundingClientRect()
          props.onUpdateCards({ top, right, bottom, left, width, height }, props.id)
        }
      }
      window.addEventListener('resize', handleResize)
      return () => {
        window.removeEventListener('resize', handleResize)
      }
    })

    const { history, onUpdateCards, id, ...rest } = props
    const goToElement = () => {
      // const { top, right, bottom, left, width, height } = element.current.getBoundingClientRect()
      // props.onUpdateCards({ top, right, bottom, left, width, height }, props.id)
      // history.push({
      //   pathname: `/work/${id}`,
      //   state: {
      //     id,
      //     to: 'modal',
      //     meta: {
      //       from: { top, right, bottom, left, width, height },
      //     },
      //   },
      // })
    }
    return (
      <div ref={el => (element.current = el)}>
        <Component id={id} onClick={goToElement} />
      </div>
    )
  }
}
export default withNavigation
