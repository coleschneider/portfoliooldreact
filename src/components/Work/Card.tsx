import * as React from 'react'
import styled, { css } from 'styled-components'

interface Props {
  isOpen: boolean
  cardId: string
}

const imageOpenMixin = ({ isOpen }: Props) =>
  isOpen &&
  css`
    opacity: 0;
    transform: translate3d(0px, -40px, 0px);
  `
const borderOpenMxin = ({ isOpen }: Props) =>
  isOpen &&
  css`
    opacity: 0;
    transform: scale(1.3);
  `

const Container = styled.div<Props>`
  width: 92%;
  max-width: 340px;
  margin-left: auto;
  margin-right: auto;
  transition: opacity 200ms linear 320ms, transform 200ms ease-out 320ms;
  position: relative;
  background: #eb5160;
  color: #fff;
  cursor: pointer;
  margin-bottom: 60px;
`
const Border = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  padding: 6px;
  border: 1px solid #fff;
  opacity: 0.5;
  left: -6px;
  top: -6px;
  ${borderOpenMxin}
`

const CardImage = styled.img`
  width: 90%;
  position: absolute;
  top: -6%;
  left: -6%;
  ${imageOpenMixin}
`
const CardText = styled.h1`
  position: relative;
  padding: 190px 0px 100px 10px;
  width: 90%;
  transform: ${({ isOpen }) => (!isOpen ? 'translate3d(20%, 0px, 0px)' : 'translate3d(20%, -40px, 0px)')};
  opacity: ${({ isOpen }) => (!isOpen ? 1 : 0)};
  transition: opacity 200ms linear 120ms, transform 200ms ease-in 120ms;
`

function Card({ isOpen, cardId, setFirstRef, handleClick }: Props) {
  return (
    <Container ref={setFirstRef} onClick={handleClick}>
      <Border isOpen={isOpen === cardId} />
      <CardImage isOpen={isOpen === cardId} src="https://www.blend.com/wp-content/uploads/2017/06/blend-logotype.png" />
      <CardText isOpen={isOpen === cardId}>Hey now, you're an allstar</CardText>
    </Container>
  )
}

export default Card
