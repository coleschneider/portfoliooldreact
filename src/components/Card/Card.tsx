/* eslint-disable react-hooks/exhaustive-deps */
import * as React from 'react'
import styled from 'styled-components'
import { animated, useTrail, SpringConfig } from 'react-spring'
import { RouteComponentProps } from 'react-router'
import withNavigation from './withNavigation'
import { Pane } from '../../theme/Elements'
import { springs } from '../../theme/animations'
import useResizeObserver from '../../hooks/useResizeObserver'
import useLazyImage from '../../hooks/useLazyImage'
import usePrevious from '../../hooks/usePrevious'
import useDimensions from '../../hooks/useMeasure/useMeasure'

export const CardWrapper = styled(Pane)`
  z-index: 1200;
`
export const CardImage = styled.img`
  width: 100%;
  object-fit: cover;
  /* z-index: 0; */
`
const ColumnFlex = styled.div`
  margin: 16px;
  flex: 1 1 250px;
`

type Props = RouteComponentProps & Card & { onUpdateCards: DimensionCallback }

interface TrailAnimation {
  config: SpringConfig
  opacity: number
  x: number
  height: number
}

function getDimensionObject(node: HTMLElement): DimensionObject {
  const { top, right, bottom, left, width, height } = node.getBoundingClientRect()

  // return {
  //   width: rect.width,
  //   height: rect.height,
  //   top: 'x' in rect ? rect.x : rect.top,
  //   left: 'y' in rect ? rect.y : rect.left,
  //   x: 'x' in rect ? rect.x : rect.left,
  //   y: 'y' in rect ? rect.y : rect.top,
  //   right: rect.right,
  //   bottom: rect.bottom,
  // }
  return { top, right, bottom, left, width, height }
}

function Card({
  onUnselectCard,
  onSelectCard,
  onUpdateCards,
  location,
  history,
  id,
  cardImage,
  description,
  placeholder,
}: Props) {
  const element = React.useRef(null)
  React.useEffect(() => {
    const handleResize = () => {
      if (element.current) {
        const { top, right, bottom, left, width, height } = element.current.getBoundingClientRect()
        onUpdateCards({ top, right, bottom, left, width, height }, id)
      }
    }
    window.addEventListener('resize', handleResize)
    return () => {
      window.removeEventListener('resize', handleResize)
    }
  }, [])

  // const imageSrc = useLazyImage(cardImage, placeholder)
  // const [node, setNode] = React.useState(null)

  const [trail, set] = useTrail<TrailAnimation>(description.length, springs.trailCards)

  // React.useEffect(() => {
  //   set({
  //     opacity: 1,
  //     x: 0,
  //     height: 80,
  //   })
  // }, [location, set])

  const animateOnClick = () => {
    const { top, right, bottom, left, width, height } = element.current.getBoundingClientRect()
    onSelectCard(id)
    onUpdateCards({ top, right, bottom, left, width, height }, id)
    history.push({
      pathname: `/work/${id}`,
      state: {
        id,
        to: 'modal',
        background: location,
        meta: {
          from: { top, right, bottom, left, width, height },
        },
      },
    })

    // selectCard(id)
    // const completed = []
    // set({
    //   opacity: 0,
    //   x: -200,
    //   height: 80,
    //   onRest: item => {
    //     completed.push(item)
    //     if (completed.length === description.length) {
    //       // eslint-disable-next-line
    //       handleClick()
    //     }
    //   },
    // })
  }
  return (
    // <ColumnFlex ref={ref} onClick={animateOnClick}>
    <CardWrapper onClick={animateOnClick} hover ref={ref => (element.current = ref)}>
      <CardImage src={cardImage} />
    </CardWrapper>
    // </ColumnFlex>
  )
}

// class Card extends React.Component {
//   element = null
//         componentDidMount(){
//             window.addEventListener('resize', this.handleResize)
//         }
//         componentWillUnmount(){
//             window.removeEventListener('resize', this.handleResize)
//         }
//         handleResize = (e) => {
//             const { top, right, bottom, left, width, height } = this.element.getBoundingClientRect();
//             this.props.onUpdateCards({ top, right, bottom, left, width, height }, this.props.id)
//             // if(this.props.isModal){
//             //   this.props.onUpdateCards({ top, right, bottom, left, width, height }, this.props.id)
//             // }

//         }
//   render(){
//   const animateOnClick = () => {

//     const { top, right, bottom, left, width, height } = this.element.getBoundingClientRect();

//     this.props.onSelectCard({ top, right, bottom, left, width, height }, this.props.id)
//     console.log(this.props)
//     this.props.history.push({
//       pathname: `/work/${this.props.id}`,
//       state: {
//         background: this.props.location,
//         to: 'modal',
//         meta: {
//           from: { top, right, bottom, left, width, height },
//         },
//       },
//     })
//   }
//     return (

//     <CardWrapper ref={ref => this.element = ref} onClick={animateOnClick} hover>
//       <CardImage src={this.props.cardImage} />
//     </CardWrapper>
//     )
//   }
// }

export default Card
