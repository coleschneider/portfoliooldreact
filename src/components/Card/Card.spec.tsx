import React from 'react'

import Card from './Card'

describe('Card', () => {
  it('should match the snapshot', () => {
    const handleClickMock = jest.fn()
    const getRefMock = jest.fn()
    const Wrapper = shallow(
      <Card location={{}} handleClick={handleClickMock} getRef={getRefMock} />
    )
    expect(Wrapper).toMatchSnapshot()
  })
  it('should call the get ref function', () => {
    const handleClickMock = jest.fn()
    const getRefMock = jest.fn()
    const Wrapper = mount(<Card location={{}} handleClick={handleClickMock} getRef={getRefMock} />)
    expect(getRefMock).toBeCalled()
  })
})
