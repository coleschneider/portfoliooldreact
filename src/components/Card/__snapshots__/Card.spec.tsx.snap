// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<Card___StyledCardWrapper
  hover={true}
  level={0}
  onClick={[Function]}
>
  <Card__ImagePopUp>
    <ForwardRef
      key="Software "
      style={
        Object {
          "color": "white",
          "fontSize": "1.5em",
          "opacity": AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 1,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 1,
            "startTime": 1571546403272,
            "value": 1,
          },
          "transform": AnimatedInterpolation {
            "attach": [Function],
            "calc": [Function],
            "children": Array [],
            "detach": [Function],
            "payload": Array [
              AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1571546403272,
                "value": 0,
              },
            ],
          },
          "whiteSpace": "pre",
        }
      }
    >
      <ForwardRef
        style={
          Object {
            "height": AnimatedValue {
              "animatedStyles": Set {},
              "children": Array [],
              "done": false,
              "lastPosition": 80,
              "lastTime": undefined,
              "lastVelocity": undefined,
              "payload": undefined,
              "setValue": [Function],
              "startPosition": 80,
              "startTime": 1571546403272,
              "value": 80,
            },
          }
        }
      >
        Software 
      </ForwardRef>
    </ForwardRef>
    <ForwardRef
      key="Engineer"
      style={
        Object {
          "color": "white",
          "fontSize": "1.5em",
          "opacity": AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 1,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 1,
            "startTime": 1571546403272,
            "value": 1,
          },
          "transform": AnimatedInterpolation {
            "attach": [Function],
            "calc": [Function],
            "children": Array [],
            "detach": [Function],
            "payload": Array [
              AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1571546403272,
                "value": 0,
              },
            ],
          },
          "whiteSpace": "pre",
        }
      }
    >
      <ForwardRef
        style={
          Object {
            "height": AnimatedValue {
              "animatedStyles": Set {},
              "children": Array [],
              "done": false,
              "lastPosition": 80,
              "lastTime": undefined,
              "lastVelocity": undefined,
              "payload": undefined,
              "setValue": [Function],
              "startPosition": 80,
              "startTime": 1571546403272,
              "value": 80,
            },
          }
        }
      >
        Engineer
      </ForwardRef>
    </ForwardRef>
  </Card__ImagePopUp>
</Card___StyledCardWrapper>
`;
